File: server/router/mine/mine_sentences.go
package mine

import (
	"github.com/gin-gonic/gin"
	"github.com/springbear2020/self-hub/server/middleware"
)

type MineSentencesRouter struct{}

func (s *MineSentencesRouter) InitMineSentencesRouter(Router *gin.RouterGroup, PublicRouter *gin.RouterGroup) {
	mineSentencesRouter := Router.Group("mineSentences").Use(middleware.OperationRecord())
	mineSentencesRouterWithoutRecord := Router.Group("mineSentences")
	mineSentencesRouterWithoutAuth := PublicRouter.Group("mineSentences")
	{
		mineSentencesRouter.POST("createMineSentences", mineSentencesApi.CreateMineSentences)
		mineSentencesRouter.DELETE("deleteMineSentences", mineSentencesApi.DeleteMineSentences)
		mineSentencesRouter.DELETE("deleteMineSentencesByIds", mineSentencesApi.DeleteMineSentencesByIds)
		mineSentencesRouter.PUT("updateMineSentences", mineSentencesApi.UpdateMineSentences)
	}
	{
		mineSentencesRouterWithoutRecord.GET("findMineSentences", mineSentencesApi.FindMineSentences)
		mineSentencesRouterWithoutRecord.GET("getMineSentencesList", mineSentencesApi.GetMineSentencesList)
		mineSentencesRouterWithoutRecord.GET("getMineSentencesStat", mineSentencesApi.GetMineSentencesStat)
	}
	{
		_ = mineSentencesRouterWithoutAuth
	}
}

File: web/src/api/mine/mine_sentences.js
import service from '@/utils/request'

export const createMineSentences = (data) => {
  return service({
    url: '/mineSentences/createMineSentences',
    method: 'post',
    data
  })
}

export const deleteMineSentences = (params) => {
  return service({
    url: '/mineSentences/deleteMineSentences',
    method: 'delete',
    params
  })
}

export const deleteMineSentencesByIds = (params) => {
  return service({
    url: '/mineSentences/deleteMineSentencesByIds',
    method: 'delete',
    params
  })
}

export const updateMineSentences = (data) => {
  return service({
    url: '/mineSentences/updateMineSentences',
    method: 'put',
    data
  })
}

export const findMineSentences = (params) => {
  return service({
    url: '/mineSentences/findMineSentences',
    method: 'get',
    params
  })
}

export const getMineSentencesList = (params) => {
  return service({
    url: '/mineSentences/getMineSentencesList',
    method: 'get',
    params
  })
}

export const getMineSentencesStat = () => {
  return service({
    url: '/mineSentences/getMineSentencesStat',
    method: 'get'
  })
}
